rostime/include/ros/impl/duration.h

```diff
@@ -52,26 +52,30 @@ namespace ros {
   template<class T>
   T& DurationBase<T>::fromSec(double d)
   {
-    sec = (int32_t)floor(d);
+    int64_t sec64 = (int64_t)floor(d);
+    if (sec64 < INT_MIN || sec64 > INT_MAX)
```

Comment:
I agree using `(U)INT32_MIN|MAX` would indeed be nicer. I kept it with `(U)INT_MIN|MAX` for now since those are already being used within the code. I would rather not mix them and don't want to update the existing code.